assets/src/hooks/useShapes.ts

```diff
@@ -30,18 +37,31 @@ export const useRouteShapes = (selectedRouteIds: RouteId[]): Shape[] => {
   useEffect(() => {
     selectedRouteIds.forEach((routeId: RouteId) => {
       if (!(routeId in shapesByRouteId)) {
-        setLoadingShapesForRoute(routeId)
+        if (!isASubwayRoute(routeId)) {
+          setLoadingShapesForRoute(routeId)
 
-        if (isASubwayRoute(routeId)) {
-          setShapesForRoute(routeId, subwayRouteShapes(routeId))
-        } else {
           fetchShapeForRoute(routeId).then((shapes: Shape[]) =>
             setShapesForRoute(routeId, shapes)
           )
+        } else if (stations) {
+          setLoadingShapesForRoute(routeId)
```

Comment:
It was intentional - basically, if `isASubwayRoute` returns true but the `stations` aren't filled in yet, then the `useEffect` doesn't actually do anything so I figured it would be best to wait on setting the loading state (`null`) until the call to the hook in which we're actually going to fire the API request.